src/main/java/com/minecolonies/coremod/entity/citizen/EntityCitizen.java

```diff
@@ -1069,7 +1069,7 @@ private boolean determineDesiredActivity()
             citizenStatusHandler.setLatestStatus(new TranslationTextComponent("com.minecolonies.coremod.status.waiting"),
               new TranslationTextComponent("com.minecolonies.coremod.status.rainStop"));
             setVisibleStatusIfNone(BAD_WEATHER);
-            desiredActivity = DesiredActivity.IDLE;
+            desiredActivity = DesiredActivity.SLEEP;
```

Comment:
Are we sure this is enough? Sleeping has recovery code in onWakeup, Do we have confirmation that onWakeup gets called when the rain stops?

Reply:
From a quick glance it doesn't seem to get called when a raid ends either

Reply:
ye onwakeup only happens when we actually did sleep, we're just reusing the sleeping AI atm for go home cases aswell, so it'll go home if its not sleep time/not sleepd then